// devexpress-reporting/viewer/widgets/_groupEditor.js
/**
* DevExpress HTML/JS Reporting (viewer\widgets\_groupEditor.js)
* Version:  22.2.3
* Build date: Dec 6, 2022
* Copyright (c) 2012 - 2022 Developer Express Inc. ALL RIGHTS RESERVED
* License: https://www.devexpress.com/Support/EULAs/universal.xml
*/
import { PropertyGridEditor } from '@devexpress/analytics-core/analytics-widgets';
export class ParametersGroupEditor extends PropertyGridEditor {
    constructor(info, level, parentDisabled, textToSearch) {
        super(info, level, parentDisabled, textToSearch);
        this.isGroupLabel = true;
    }
    _setPadding(position, value) {
        var padding = super._setPadding(position, value);
        padding['padding-' + position] = 10;
        padding['padding-bottom'] = 6;
        return padding;
    }
    createObjectProperties() {
        this.collapsed(!this.editorOptions.expanded && this.editorOptions.titleVisible);
        this.hideEditorHeader = !this.editorOptions.titleVisible;
        this.hideCollapsingImage = !this.editorOptions.showExpandButton;
        this.hideBorder = this.editorOptions.borderVisible === false;
        this.showHorizontally = this.editorOptions.orientation === 'Horizontal';
        this.level = -1;
        this.editorCreated(true);
        return super.createObjectProperties();
    }
}
